packages/core/src/internal/proc.js

```diff
@@ -294,7 +294,8 @@ export default function proc(env, iterator, parentContext, parentEffectId, meta,
 
         if (env.onError) {
           env.onError(result)
-        } else {
+        }
```

Comment:
note: that's actually a change in api.

Reply:
seems to me that we dont need both `onError` & `logError`, having them both is confusing

Reply:
actually use case for them a bit different (at least now) 
1 `onError` - called once, when error bubbles to the root, provides info about error source. usually it's used in user land to get source of error and report them (logger is just another target!)
2 `logError` - can be called several times, called for an error bubbling to the root saga + internal inconsistency errors (when try to cancel already cancelled task).

probably we should add a note about their different nature or remove one of them, but on the whole it seems that they have different use cases

Reply:
> updated: there is another problem with onError we build saga error stack only when error bubbles to the root, so if user handles errors on their own (say try/catch on the upper layer), they don't have normalized saga error stack

But that's OK I think. People shouldnt rely on the stack anyhow - it doesnt matter if we are talking about normal errors or saga errors, stack inspection is shady and unreliable. 

The whole point of saga stack is to improve developer experience because errors caught by saga are otherwise really cryptic. No runtime code should rely on them though.

I think we should get rid of `logError` entirely and just recommend using `onError` for logging purposes. Explaining in the docs that it's role is primarily for logging purposes and it's called only for entirely uncaught errors from which we couldnt recover.

Reply:
@Andarist ok, what about saga internal inconsistency errors? don't log them?

Reply:
could we check how other libraries handle such errors? (ie. react)

Reply:
react throws errors with https://github.com/facebook/react/blob/master/packages/shared/invariant.js

example with usage error:
https://github.com/facebook/react/blob/master/packages/react-dom/src/client/ReactDOMTextarea.js#L44-L47
example with internal error:
https://github.com/facebook/react/blob/master/packages/shared/ReactErrorUtils.js#L120-L124

Reply:
so basically both are handled the same way, i think we can safely assume this is what people expect, so both type of errors should just get handled to `onError` or logged with `console.error` if `onError` is not provided

Reply:
there at least 2 kinds of errors in React
- in user component. can be handled by `componentDidCatch`. user can recover from error, report etc.
- internal inconsistency in React. thrown as exception. user cannot handle it.

for me redux-saga `onError` hook is similar to `componentDidCatch`. not quite sure we should mix approaches. I'd throw an exception in case of internal saga error

Reply:
ok, let's continue discussion in the related  issue. I'd merge PR